### PR TITLE
BUG 1684206: *: store etcd CA and client certs in cluster

### PR DESCRIPTION
**Pull request by abhinavdahiya**
xref: https://bugzilla.redhat.com/show_bug.cgi?id=1684206

This pushes the etcd ca into cluster to sign future etcd client CSR requests.
/cc @csrwng

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
rebase around #1291

File: data/data/manifests/bootkube/kube-system-secret-etcd-client-ca-deprecated.yaml.template [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: etcd-client-ca-deprecated
```
Comment (wking) [inline data/data/manifests/bootkube/kube-system-secret-etcd-client-ca-deprecated.yaml.template]: Why are we opening with a deprecated object?

File: pkg/asset/templates/content/bootkube/pull.go [hunk 2]
```diff
@@ -9,12 +9,12 @@ import (
 )
 
 const (
-	pullFileName = "pull.yaml.template"
+	pullFileName = "pull.json.template"
```
Comment (wking) [inline pkg/asset/templates/content/bootkube/pull.go]: This seems unrelated?

Comment (wking) [reply pkg/asset/templates/content/bootkube/pull.go]: Ah, I see from the commit message:

> fixes the template for pull as the contents were json but file was names yaml

But JSON is a subset of YAML, so `.yaml` was accurate, but less specific.  Anyhow, I'm fine with this rename if you don't want to bother splitting it out ;).

File: data/data/manifests/bootkube/kube-system-secret-etcd-client-ca-deprecated.yaml.template (hunk 1 above)
Comment (abhinavdahiya) [reply data/data/manifests/bootkube/kube-system-secret-etcd-client-ca-deprecated.yaml.template]: > Why are we opening with a deprecated object?

no sure what you mean by that.

This mimics the deprecation comment on https://github.com/openshift/installer/blob/c52894b2c2064e5cb8a6d736ccdbe29279d18ceb/pkg/asset/tls/etcd.go#L11
anf https://github.com/openshift/installer/blob/c52894b2c2064e5cb8a6d736ccdbe29279d18ceb/pkg/asset/manifests/operators.go#L168

File: pkg/asset/templates/content/bootkube/pull.go (hunk 2 above)
Comment (abhinavdahiya) [reply pkg/asset/templates/content/bootkube/pull.go]: I have mentioned that in https://github.com/openshift/installer/pull/1363/commits/c52894b2c2064e5cb8a6d736ccdbe29279d18ceb
```
modifies the manifest target to use the filename from template assets, rather than duplicating the filename.

fixes the template for pull as the contents were json but file was names yaml
```

Comment (abhinavdahiya) [reply pkg/asset/templates/content/bootkube/pull.go]: so mco requires the file name to be [`pull.json`](https://github.com/openshift/installer/blob/ade95dd3cbb88efc6afe601dadce8e7977bc3053/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L178)

and when I was making the change it got confused wriring the write file name to write asset.... :( so I made the change  https://github.com/openshift/installer/commit/c52894b2c2064e5cb8a6d736ccdbe29279d18ceb

File: pkg/asset/manifests/operators.go [hunk 3]
```diff
@@ -150,65 +162,52 @@ func (m *Manifests) generateBootKubeManifests(dependencies asset.Parents) []*ass
...
+		&bootkube.KubeSystemSecretEtcdSignerClient{},
+		&bootkube.MachineConfigServerTLSSecret{},
+		&bootkube.OpenshiftConfigConfigmapEtcdMetricsServingCA{},
+		&bootkube.OpenshiftConfigSecretEtcdMetricsClient{},
+		&bootkube.OpenshiftMachineConfigOperator{},
+		&bootkube.Pull{},
+	} {
+		dependencies.Get(a)
+		for _, f := range a.Files() {
+			assetData[strings.TrimSuffix(filepath.Base(f.Filename), ".template")] = applyTemplateData(f.Data, templateData)
```
Comment (staebler) [inline pkg/asset/manifests/operators.go]: This is a welcome change over what was being done. We don't need the `assetData` map as a temporary home. We can add the file directly to a `*assetFile` slice here.

File: pkg/asset/templates/content/bootkube/pull.go (hunk 2 above)
Comment (staebler) [reply pkg/asset/templates/content/bootkube/pull.go]: +1 for consistency between the name of the template and the name of the file generated.

File: pkg/asset/manifests/operators.go (hunk 3 above)
Comment (abhinavdahiya) [reply pkg/asset/manifests/operators.go]: donw with https://github.com/openshift/installer/compare/c52894b2c2064e5cb8a6d736ccdbe29279d18ceb..53c6fc3c1af4567bff062424a0a5aef6d587d93f
